content/en/blog/_posts/2019-12-09-volume-snapshot-beta.md

```diff
@@ -0,0 +1,449 @@
+
+If no `volumeSnapshotClassName` is specified, one is automatically selected as follows:
+The `StorageClass` from PVC or PV of the source volume is fetched.
+The default `VolumeSnapshotClass` is fetched, if available. A default `VolumeSnapshotClass` is a snapshot class created by the admin with the `snapshot.storage.kubernetes.io/is-default-class` annotation. If the `Driver` field of the default `VolumeSnapshotClass` is the same as the `Provisioner` field in the `StorageClass`, the default `VolumeSnapshotClass` is used. If there is no default `VolumeSnapshotClass` or more than one default `VolumeSnapshotClass` for a snapshot, an error will be returned.
+
+Please note that the Kubernetes Snapshot API does not provide any consistency guarantees. You have to prepare your application (pause application, freeze filesystem etc.) before taking the snapshot for data consistency either manually or using some other higher level APIs/controllers.
+
+You can verify that the `VolumeSnapshot` object is created and bound with `VolumeSnapshotContent` by running `kubectl describe volumesnapshot`:
+
+`Bound Volume Snapshot Content Name` - field in the `Status` indicates indicates the volume is bound to the specified `VolumeSnapshotContent`.
+`Ready To Use` - field in the `Status` indicates this volume snapshot is ready for use.
+`Creation Time` - field in the `Status` indicates when the snapshot was actually created (cut).
+`Restore Size` - field in the `Status` indicates the minimum volume size required when restoring a volume from this snapshot.
+
+```
+Name:         test-snapshot
+Namespace:    default
+Labels:       <none>
+Annotations:  <none>
+API Version:  snapshot.storage.k8s.io/v1beta1
+Kind:         VolumeSnapshot
+Metadata:
+  Creation Timestamp:  2019-11-16T00:36:04Z
+  Finalizers:
+    snapshot.storage.kubernetes.io/volumesnapshot-as-source-protection
+    snapshot.storage.kubernetes.io/volumesnapshot-bound-protection
+  Generation:        1
+  Resource Version:  1294
+  Self Link:         /apis/snapshot.storage.k8s.io/v1beta1/namespaces/default/volumesnapshots/new-snapshot-demo
+  UID:               32ceaa2a-3802-4edd-a808-58c4f1bd7869
+Spec:
+  Source:
+    Persistent Volume Claim Name:  test-pvc
+  Volume Snapshot Class Name:      test-snapclass
+Status:
+  Bound Volume Snapshot Content Name:  snapcontent-32ceaa2a-3802-4edd-a808-58c4f1bd7869
+  Creation Time:                       2019-11-16T00:36:04Z
+  Ready To Use:                        true
+  Restore Size:                        1Gi
+```
+
+As a reminder to any developers building controllers using volume snapshot APIs: before using a `VolumeSnapshot` API object, validate the bi-directional binding between the `VolumeSnpashot` and the `VolumeSnapshotContent` it is bound to, to ensure the binding is complete and correct (not doing so may result in security issues).
+
+```shell
+kubectl describe volumesnapshotcontent
+
```

Comment:
It's best to separate code from sample output (replace this one blank line with two lines, both containing three backticks).

Note: there's no way to put this in as a GitHub suggestion due to the way the feature works.